client/app/lib/components/RailsConfirmationDialog.jsx

```diff
@@ -0,0 +1,37 @@
+import React, { PropTypes } from 'react';
+import ConfirmationDialog from 'lib/components/ConfirmationDialog';
+
+class RailsConfirmationDialog extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { open: true, disableButtons: false };
+    this.onConfirm = this.onConfirm.bind(this);
+  }
+
+  // Disable buttons once the confirm button is clicked, then do confirm callback.
+  onConfirm() {
+    const { onConfirmCallback } = this.props;
+    this.setState({ disableButtons: true });
+    onConfirmCallback();
```

Comment:
No value in extracting out `onConfirmCallback` as a variable. Just do `this.props.onConfirmCallback()`